packages/trader/src/Modules/Contract/Containers/contract-replay.tsx

```diff
@@ -0,0 +1,384 @@
+                        >
+                            {contract_drawer_el}
+                        </div>
+                    </MobileWrapper>
+                    <React.Suspense fallback={<div />}>
+                        <div
+                            className={classNames('replay-chart__container', {
+                                'replay-chart__container--is-multiplier': isMobile() && is_multiplier,
+                                'vanilla-trade-chart': is_vanilla,
+                            })}
+                        >
+                            <DesktopWrapper>
+                                <NotificationMessages />
+                            </DesktopWrapper>
+                            <ChartLoader is_dark={is_dark_theme} is_visible={is_chart_loading} />
+                            <DesktopWrapper>
+                                <ReplayChart
+                                    is_dark_theme_prop={is_dark_theme}
+                                    is_accumulator_contract={is_accumulator}
+                                />
+                            </DesktopWrapper>
+                            <MobileWrapper>
+                                {is_digit_contract ? (
+                                    <React.Fragment>
+                                        <InfoBoxWidget />
+                                        <SwipeableWrapper
+                                            className='replay-chart__container-swipeable-wrapper'
+                                            is_swipe_disabled={swipe_index === 1}
+                                            onChange={
+                                                onChangeSwipeableIndex as React.ComponentProps<
+                                                    typeof SwipeableWrapper
+                                                >['onChange']
+                                            }
+                                        >
+                                            <DigitsWidget />
+                                            <ReplayChart />
+                                        </SwipeableWrapper>
+                                    </React.Fragment>
+                                ) : (
+                                    <ReplayChart
+                                        is_dark_theme_prop={is_dark_theme}
+                                        is_accumulator_contract={is_accumulator}
+                                    />
+                                )}
+                            </MobileWrapper>
+                        </div>
+                    </React.Suspense>
+                </Div100vhContainer>
+            </PageOverlay>
+        </FadeWrapper>
+    );
+});
+
+export default ContractReplay;
+
+// CHART -----------------------------------------
+
+const ReplayChart = observer(
+    ({
+        is_dark_theme_prop,
+        is_accumulator_contract,
+    }: {
+        is_dark_theme_prop?: boolean;
```

Comment:
Previously `is_dark_theme_prop` was not in the props, but the component received this variable + was extracting same value from `ui` store. So I decided to add it to props and implement some kind of backup:
`const is_dark_theme = is_dark_theme_prop || ui.is_dark_mode_on;`